### ISSUE TITLE
Add Drush launcher function to examples/example.bashrc

### ISSUE THREAD

**Issue by Chi-teck** [drush-ops/drush#2677]
The documentation states that on D8 sites Drush should be installed locally through Composer.

Here is an example (not much tested) of such a function. Unlike _drush.wrapper_ this allows to start Drush from any directory within Drupal 8 installation.

```bash
# Drush wrapper.
function drush {
  local current_dir=$(pwd)
  while true; do
    # Nothing.
    if [ $current_dir == '/' ]; then
      echo Could not find a Drupal site to operate on.
      return 1;
    # Drupal 8.
    elif [ -f $current_dir/index.php ] && [ -f $current_dir/core/includes/bootstrap.inc ]; then
      if [ -f $current_dir/vendor/bin/drush ]; then
        $current_dir/vendor/bin/drush $@
        return $?
      else
        echo Drush is not installed on this Drupal site.
        return 1
      fi
    # Drupal 7.
    elif [ -f $current_dir/index.php ] && [ -f $current_dir/includes/bootstrap.inc ]; then
      # Launch global Drush.
      /usr/local/bin/drush $@
      return $?
    fi
    current_dir=$(dirname $current_dir)
  done
}
```

**Comment by weitzman** [comment]
See another attempt at [drush-shim](https://github.com/webflo/drush-shim). Your thoughts on how to proceed are most appreciated.

**Comment by Chi-teck** [comment]
I tested _drush-shim_ before writing this function. It basically worked the same way.  This approach  seems a bit simpler and more customizable for me.

The main challenge in both cases is how to deal with Drupal 6, 7 sites and how to support global commands that need no Drupal context (like `drush help`).

Basically my thinking about this is as follows:

1. Provide an LTS release for legacy sites which can only be installed globally (composer global, phar file, whatever).
2. Provide a release (composer package) for Drupal 8 sites which is installed locally (this should not be a recommendation like now but a requirement).
3. Optionally provide a way (shim) to select appropriate Drush executable automatically based on directory context.

The great benefit of this is that Drush will not have to support multiple major Drupal versions withing the same release. This can significantly improve a maintainability of the project.

**Comment by weitzman** [comment]
You thinking aligns with mine. Drush9 will only support Drupal 8 as per https://medium.com/@weitzman/drush-9-roadmap-9eb5cd227c50#.44ex17a3k. I think we can remove the Drupal 7 handling in this PR. I hope to test the script soon.

**Comment by weitzman** [comment]
@Chi-teck - I have been thinking of adding a dependency in Drush to DrupalFinder. See how it will is being used at https://github.com/drupal-composer/drupal-project/commit/9f84dac. Would that be helpful for the drush.wrapper?

**Comment by Chi-teck** [comment]
I think It wouldn't be unless _drush.wrapper_ is installed globally.

**Comment by weitzman** [comment]
Fixed by #2749
